lib/src/mock_query.dart

```diff
@@ -80,13 +80,21 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
 
           dynamic value1;
           if (field is String) {
-            value1 = d1.get(field) as Comparable;
+            try {
+              value1 = d1.get(field) as Comparable;
+            } catch (error) {
```

Comment:
This catch catches the case when the key/value does not exist and the case when the value is null, and as a result not a Comparable. A comment would help future readers what is happening.